### PR TITLE
Support string search operator for Enum data type

### LINKED ISSUE #72661: Support LIKE operator for Enum data type
**Use case**

LIKE/ILIKE should work no matter what data type is in the column.

**Describe the solution you'd like**

LIKE will search in string representation of Enum data type values.

Basically `enum_column::String LIKE '%abc%'` works and `enum_column LIKE '%abc%'` does not.

### PR DESCRIPTION
**Pull request by zhanglistar**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Support string search operator(eg. like) for Enum data type, fix https://github.com/ClickHouse/ClickHouse/issues/72661.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 7d0a62e351f5743c849ea73413df73356315ac20 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/72732/7d0a62e351f5743c849ea73413df73356315ac20/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/b81ee27750c4520de6e6b979ce7d4f6a111cb8ea/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: tests/queries/0_stateless/03278_enum_string_functions.sql [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+-- countSburstringsCaseInsensitive --
+SELECT countSubstringsCaseInsensitive(e, 'a') FROM test_enum_string_functions;
+-- countSubstringsCaseInsensitiveUTF8 --
+SELECT countSubstringsCaseInsensitiveUTF8(e, 'a') FROM test_enum_string_functions;
+-- hasToken --
+SELECT hasToken(e, 'a') FROM test_enum_string_functions;
+-- hasTokenOrNull --
+SELECT hasTokenOrNull(e, 'a') FROM test_enum_string_functions;
+
+DROP TABLE jsons;
```
Comment (vdimir) [inline tests/queries/0_stateless/03278_enum_string_functions.sql:27]:
```suggestion
DROP TABLE IF EXISTS jsons;
```

**Comment by qoega** [comment]
You need order by for stable test result?
https://s3.amazonaws.com/clickhouse-test-reports/72732/bb7d7add4607043953c96042d00b84061d89f81b/stateless_tests_flaky_check__asan_.html

**Comment by zhanglistar** [comment]
> You need order by for stable test result? https://s3.amazonaws.com/clickhouse-test-reports/72732/bb7d7add4607043953c96042d00b84061d89f81b/stateless_tests_flaky_check__asan_.html

Yes, right, thanks for pointing it out.

**Review by vdimir — APPROVED** [review]
👍

**Pull request merged**